src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -1152,7 +1231,17 @@ def _validate_child(position, child):
         return False
 
     def gen_code(self, parent):
+        '''
+        Generate the f2pygen AST entries in the Schedule for this OpenMP
+        directive.
+        Note gen_code ignores the collapse clause but the generated code is
+        still valid.
```

Comment:
Again, please add a note about when/how this restriction will be lifted. This ought to be documented in the relevant transformation as well as I can imagine a user really struggling with why their requested collapse hasn't appeared in the generated code.

Reply:
Added TODO here.

The OMP transformation does not have an option to request a collapse directive. When needed I am doing it manually after inserting the directive using the node.collapse property setter, so I added another TODO in the property setter.

Reply:
Aha. I feel happier about that then. Thanks.